src/field_theory/krull_topology.lean

```diff
@@ -283,3 +285,242 @@ begin
+on `L ≃ₐ[K] L`. Then the restriction map `(L ≃ₐ[K] L) → (E →ₐ[K] L)` pushes `f` forward to an
+ultrafilter on `E →ₐ[K] L`. Since `E →ₐ[K] L` is a finite set, this ultrafilter is principal. The
+element of `E →ₐ[K] L` generating this principal ultrafilter is `f.alg_hom h_findim`, where
+`h_findim : finite_dimensional K E`. -/
+protected noncomputable def ultrafilter.generator_of_pushforward
+  (f : ultrafilter (L →ₐ[K] L)) {E : Type*} [field E] [algebra K E] [algebra E L]
+  [is_scalar_tower K E L] [finite_dimensional K E]: E →ₐ[K] L :=
+classical.some $ ultrafilter.eq_pure_of_fintype $
+  f.map $ λ σ, σ.comp $ is_scalar_tower.to_alg_hom K E L
+
+/-- Let `f` be an ultrafilter on `L ≃ₐ[K] L`. For an intermediate field `E` of `L/K`, there is a
+  natural restriction map `(L ≃ₐ[K] L) → (E →ₐ[K] L)`. Moreover, this restriction map gives
+  a pushforward of `f` to an ultrafilter on `E →ₐ[K] L`. If `E/K` is finite, then this pushforward
+  is generated by `ultrafilter.alg_hom h_findim f`.  -/
+lemma ultrafilter.generator_of_pushforward_spec (f : ultrafilter (L →ₐ[K] L)) {E : Type*} [field E]
+  [algebra K E] [algebra E L] [is_scalar_tower K E L] [finite_dimensional K E]:
+  f.map (λ σ : L →ₐ[K] L, σ.comp $ is_scalar_tower.to_alg_hom K E L) =
+  pure (f.generator_of_pushforward) :=
+coe_injective $ classical.some_spec $ ultrafilter.eq_pure_of_fintype _
+
+
+lemma ultrafilter.generator_of_pushforward_comp_inclusion (f : ultrafilter (L →ₐ[K] L))
+  (hE : finite_dimensional K E) (hF : finite_dimensional K F) (hEF : E ≤ F) :
+  (f.generator_of_pushforward : F →ₐ[K] L).comp (subalgebra.inclusion hEF) =
+  (f.generator_of_pushforward : E →ₐ[K] L) :=
+begin
+  refine ultrafilter.pure_injective _,
+  transitivity map (λ ϕ : ↥F →ₐ[K] L, ϕ.comp $ subalgebra.inclusion hEF)
+  (pure $ f.generator_of_pushforward),
+  { refl },
+  rw [←f.generator_of_pushforward_spec, ←f.generator_of_pushforward_spec],
+  refl,
+end
+
+/-- For each `x : L`, the intermediate field `K(x)` is finite dimensional over `K`, so
+`f.alg_hom` gives a map `ϕₓ : K(x) → L`. We define
+`f.glued_generators_of_pushforwards_function h_int` to be the function taking `x`
+to `ϕₓ(x)` for all `x : L`. -/
+protected noncomputable def ultrafilter.glued_generators_of_pushforwards_function
+  (h_int : algebra.is_integral K L) (f : ultrafilter (L →ₐ[K] L)) (x : L) : L :=
+      let foo : finite_dimensional K (intermediate_field.adjoin K {x}) :=
+      intermediate_field.adjoin.finite_dimensional (h_int x) in
+    by exactI (f.generator_of_pushforward : (intermediate_field.adjoin K {x}) →ₐ[K] L)
+  ⟨x, intermediate_field.mem_adjoin_simple_self K x⟩
+
+lemma ultrafilter.glued_generators_of_pushforwards_function_spec (h_int : algebra.is_integral K L)
+  (f : ultrafilter (L →ₐ[K] L)) (hE : finite_dimensional K E) {x : L} (hx : x ∈ E) :
+  f.glued_generators_of_pushforwards_function h_int x =
+  (f.generator_of_pushforward : (E →ₐ[K] L)) ⟨x, hx⟩ :=
+  begin
```

Comment:
Indents are weird here still: the first line after `:=` should never be indented. I think you fixed this earlier (based on https://github.com/leanprover-community/mathlib/pull/13307/files#r880836759), but then undid it again by accident.